Extensions/DialogueTree/dialoguetools.js

```diff
@@ -0,0 +1,583 @@
+/**
+ * Use this to get the current complete dialogue line without using any scrolling effects.
+ * Note that using this instead getClippedLineText will skip any <<wait>> commands entirely.
+ */
+gdjs.dialogueTree.getLineText = function() {
+  this.clipTextEnd = this.dialogueText.length;
+  return this.dialogueText.length ? this.dialogueText : "";
+};
+
+/**
+ * use this to get the number of command parameters in a command with parameters that has been caught by a isCommandCalled condition
+ */
+gdjs.dialogueTree.commandParametersCount = function() {
+  if (this.commandParameters && this.commandParameters.length > 1) {
+    return this.commandParameters.length - 1;
+  }
+  return 0;
+};
+
+/**
+ * use this to get a command parameter in any command with parameters that has been caugth by a isCommandCalled condition
+ * @param {number} paramIndex The index of the parameter you want to get. If left empty, the expression will get the first parameter by default.
+ */
+gdjs.dialogueTree.getCommandParameter = function(paramIndex) {
+  if (
+    this.commandParameters &&
+    this.commandParameters.length >= paramIndex + 1
+  ) {
+    var returnedParam = this.commandParameters[paramIndex + 1];
+    return returnedParam ? returnedParam : "";
+  }
+  return "";
+};
+
+/**
+ * use this to catch <<commands>> and <<commands with parameters>> from the current Dialogue Line.
+ * Use this to trigger events in the event sheet that relate to the story you are telling throught the dialogue.
+ * @param {string} command The command you want to check for being called. Write it without the <<>>
+ */
+gdjs.dialogueTree.isCommandCalled = function(command) {
+  var { commandCalls, clipTextEnd, dialogueText } = gdjs.dialogueTree;
+
+  if (this.pauseScrolling || !commandCalls) return false;
+  return this.commandCalls.some(function(call, index) {
+    if (clipTextEnd < call.time) return false;
+    if (call.cmd === "wait" && clipTextEnd !== dialogueText.length) {
+      gdjs.dialogueTree.pauseScrolling = true;
+      setTimeout(function() {
+        gdjs.dialogueTree.pauseScrolling = false;
+        commandCalls.splice(index, 1);
+      }, parseInt(call.params[1], 10));
+    }
+    if (call.cmd === command) {
+      gdjs.dialogueTree.commandParameters = call.params;
+      commandCalls.splice(index, 1);
+      return true;
+    }
+  });
+};
+
+/**
+ * Internal method to allow for capping option selection
+ */
+gdjs.dialogueTree._normalizedOptionIndex = function(optionIndex) {
+  if (optionIndex >= this.options.length) optionIndex = this.options.length - 1;
+  if (optionIndex < 0) optionIndex = 0;
+  return optionIndex;
+};
+
+/**
+ * Internal method to allow for cycling option selection
+ */
+gdjs.dialogueTree._cycledOptionIndex = function(optionIndex) {
+  if (optionIndex >= this.options.length) optionIndex = 0;
+  if (optionIndex < 0) optionIndex = this.options.length - 1;
+  return optionIndex;
+};
+
+/**
+ * use this to get the text of an option the player can select.
```

Comment:
(same comment later) Remove all the "Use this" from all function descriptions. They just add noise to the description. Someone using JavaScript already now how to use a function. The description should state what the function does: "Get the text of an option the player can select.", optionally followed by more context/explanation (in which case "Use this with [blabla]" is fine)